data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/KinesisClientAPIHandler.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+
+package org.opensearch.dataprepper.plugins.kinesis.source;
+
+import com.amazonaws.arn.Arn;
+import com.linecorp.armeria.client.retry.Backoff;
+import lombok.extern.slf4j.Slf4j;
+import org.opensearch.dataprepper.plugins.kinesis.source.exceptions.KinesisRetriesExhaustedException;
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient;
+import software.amazon.awssdk.services.kinesis.model.DescribeStreamSummaryRequest;
+import software.amazon.awssdk.services.kinesis.model.DescribeStreamSummaryResponse;
+import software.amazon.awssdk.services.kinesis.model.StreamDescriptionSummary;
+import software.amazon.kinesis.common.StreamIdentifier;
+
+import java.time.Duration;
+
+@Slf4j
+public class KinesisClientAPIHandler {
+    private static final String COLON = ":";
+
+    private final Backoff backoff;
+    private final KinesisAsyncClient kinesisClient;
+    private int failedAttemptCount;
+    private int maxRetryCount;
+
+    public KinesisClientAPIHandler(final KinesisAsyncClient kinesisClient, final Backoff backoff, final int maxRetryCount) {
+        this.kinesisClient = kinesisClient;
+        this.backoff = backoff;
+        this.failedAttemptCount = 0;
+        if (maxRetryCount <= 0) {
+            throw new IllegalArgumentException("Maximum Retry count should be strictly greater than zero.");
+        }
+        this.maxRetryCount = maxRetryCount;
+    }
+
+    public StreamIdentifier getStreamIdentifier(final String streamName) {
+        failedAttemptCount = 0;
+        while (failedAttemptCount < maxRetryCount) {
+            try {
+                DescribeStreamSummaryRequest describeStreamSummaryRequest = DescribeStreamSummaryRequest.builder()
```

Comment:
You can move this outside of the `while` loop so that you don't have to recreate it.